build: release: v1.14.0-rc1 (#7921)
filecoin-project/lotus: CHANGELOG.md

# Lotus changelog

# 1.14.0-rc1 / 2022-01-12

This is the first release candidate for the mandatory release v1.14.0 of Lotus that introduces [Filecoin network v15, 
codenamed the OhSnap upgrade](https://github.com/filecoin-project/community/discussions/74?sort=new#discussioncomment-1922550).

> The OhSnap upgrade introduces the following FIPs, delivered in actor v7-rc1:

Reviewer: ```suggestion
The OhSnap upgrade introduces the following FIPs, delivered in [actors v7-rc1](https://github.com/filecoin-project/specs-actors/releases/tag/v7.0.0-rc1):
```

- [FIP-0019 Snap Deals](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0019.md)
- [FIP-0028 Remove Datacap from Verified clients](https://github.com/filecoin-project/FIPs/pull/226)

Note: 

> - This release is built on top of lotus v1.13.2. Enterprise users like storage providers, data brokers and others 

Reviewer: ```suggestion
- This release is built on top of lotus v1.13.2. Enterprising users like storage providers, data brokers and others 
```

  are recommended to test out v1.15.0-rc1(coming on Jan 18th, 2022) for latest new features, improvements and bug 
  fixes and be ready for the upgrade!
- This release candidate uses temporary proof params for Snap Deals.
- It only sets upgrade epoch for Butterfly-SnapNet, and it does not set the upgrade epochs for calibration and mainnet.

## Butterfly - SnapNet

The ButterFly-SnapNet will be upgraded to Network v15 OhSnap at epoch 30262, around 2022-01-17T19:00:00Z. 

> To join the network, simply building lotus by running `make butterflynet`. 

Reviewer: ```suggestion
To join the network, simply build lotus by running `make butterflynet`. 
```

> The network supports three sector sizes, 512MiB, 32GiB and 64GiB. Temporary proof params for Snap Deals should be downloaded upon your node restarts. 

Reviewer: ```suggestion
The network supports three sector sizes, 512MiB, 32GiB and 64GiB. Temporary proof params for Snap Deals should be downloaded upon your node restart. 
```

>  - The parameters are pinged on IPFS gateway https://proofs.filecoin.io/ipfs/ and the CIDs can be found [here](https://github.com/filecoin-project/lotus/blob/edd3486d2cf53b960382e9cda6671e647844aa41/build/proof-params/parameters.json), please let the lotus team know if the params are not fetched automatically. You can also download the params manually from s3://proof-params-ap/filecoin-snapdeal-parameters/.

Reviewer: Can we suggest exactly how the lotus team can be alerted in this case (eg "file an issue here" / "open a discussion here" / "ask in this Slack channel")

- Implement and support [FIP-0019 Snap Deals](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0019.md)
  - chore: deps: Integrate proof v11.0.0 ([#7923](https://github.com/filecoin-project/lotus/pull/7923))
    - Snap Deals Lotus Integration: FSM Posting and integration test ([#7810](https://github.com/filecoin-project/lotus/pull/7810))
    - Feat/sector storage unseal ([#7730](https://github.com/filecoin-project/lotus/pull/7730))
    - Feat/snap deals storage ([#7615](https://github.com/filecoin-project/lotus/pull/7615))
    - fix: sealing: Add more deal expiration checks during PRU pipeline ([#7871](https://github.com/filecoin-project/lotus/pull/7871))
    - chore: deps: Update go-paramfetch ([#7917](https://github.com/filecoin-project/lotus/pull/7917))
    - feat: #7880 gas: add gas charge for VerifyReplicaUpdate ([#7897](https://github.com/filecoin-project/lotus/pull/7897))
    - enhancement: sectors: disable existing cc upgrade path 2 days before the upgrade epoch ([#7900](https://github.com/filecoin-project/lotus/pull/7639))

## Improvements
- updating to new datastore/blockstore code with contexts ([#7646](https://github.com/filecoin-project/lotus/pull/7639))

> - reorder transfer checks so as to ensure sending 2B FIL to yourself fails if you don't have that amount ([#7637](https://github.com/filecoin-project/lotus/pull/7639))

Reviewer: ```suggestion
- VM: Rework handling of self-transfers to fail on insufficient balance ([#7637](https://github.com/filecoin-project/lotus/pull/7639))
```

Reviewer: also URL

## Bug Fixes
- Fix: state: circsuypply calc around null blocks ([#7890](https://github.com/filecoin-project/lotus/pull/7890))

> - VM: Circ supply should be constant per epoch ([#7811](https://github.com/filecoin-project/lotus/pull/7811))

Reviewer: I would move this to improvements
